src/cascadia/UnitTests_TerminalCore/SelectionTest.cpp

```diff
@@ -178,5 +178,106 @@ namespace TerminalCoreUnitTests
                 rowValue++;
             }
         }
+
+        TEST_METHOD(SelectWideGlyph_Trailing)
+        {
+            Terminal term = Terminal();
+            DummyRenderTarget emptyRT;
+            term.Create({ 100, 100 }, 0, emptyRT);
+
+            // Insert wide glyph at position (4,10)
+            term.SetCursorPosition(4, 10);
+            term.Write(L"\xF0\x9F\x98\x83");
+
+            // Simulate click at (x,y) = (5,10)
+            auto clickPos = COORD{ 5, 10 };
+            term.SetSelectionAnchor(clickPos);
+
+            // Simulate renderer calling TriggerSelection and acquiring selection area
+            auto selectionRects = term.GetSelectionRects();
+
+            // Validate selection area
+            // Selection should expand one to the left to get the leading half of the wide glyph
+            VERIFY_ARE_EQUAL(selectionRects.size(), static_cast<size_t>(1));
+
+            auto selection = term.GetViewport().ConvertToOrigin(selectionRects.at(0)).ToInclusive();
+            VerifyCompareTraits<SMALL_RECT>::AreEqual({ 4, 10, 10, 5 }, selection);
+        }
+
+        TEST_METHOD(SelectWideGlyph_Leading)
+        {
+            Terminal term = Terminal();
+            DummyRenderTarget emptyRT;
+            term.Create({ 100, 100 }, 0, emptyRT);
+
+            // Insert wide glyph at position (4,10)
+            term.SetCursorPosition(4, 10);
+            term.Write(L"\xF0\x9F\x98\x83");
+
+            // Simulate click at (x,y) = (5,10)
+            auto clickPos = COORD{ 4, 10 };
+            term.SetSelectionAnchor(clickPos);
+
+            // Simulate renderer calling TriggerSelection and acquiring selection area
+            auto selectionRects = term.GetSelectionRects();
+
+            // Validate selection area
+            // Selection should expand one to the left to get the leading half of the wide glyph
+            VERIFY_ARE_EQUAL(selectionRects.size(), static_cast<size_t>(1));
+
+            auto selection = term.GetViewport().ConvertToOrigin(selectionRects.at(0)).ToInclusive();
+            VerifyCompareTraits<SMALL_RECT>::AreEqual({ 4, 10, 10, 5 }, selection);
+        }
+
+        TEST_METHOD(SelectWideGlyphsInBoxSelection)
+        {
+            Terminal term = Terminal();
+            DummyRenderTarget emptyRT;
+            term.Create({ 100, 100 }, 0, emptyRT);
+
+            // Insert wide glyph at position (4,10)
+            term.SetCursorPosition(4, 10);
+            term.Write(L"\xF0\x9F\x98\x83");
+
+            // Insert wide glyph at position (7,11)
+            term.SetCursorPosition(7, 11);
+            term.Write(L"\xF0\x9F\x98\x83");

```

Comment:
i am very worried about this actually: this is `00F0 009F 0098 0083`, which is four otherwise normal UTF-16 characters. You might need to formulate these as UTF-16 codepoints. How is the test passing???